homeassistant/components/glances/sensor.py

```diff
@@ -34,7 +34,7 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
             elif sensor_details[0] == "sensors":
                 # sensors will provide temp for different devices
                 for sensor in client.api.data[sensor_details[0]]:
-                    if sensor["type"] == sensor_type:
+                    if sensor["type"].startswith(sensor_type):
```

Comment:
It looks like this will pickup non-temperature sensors as well but the logic below hasn't been adjusted to handle this case.

Reply:
I don't quite understand the problem here, would you please give me an example if possible?

The original `if sensor["type"] == sensor_type:` also applies to non-temperature sensors, and the new `startswith` should not break the matching.

For e.g., assume a sensor with `sensor_type` as `temperature_core`, in previous version it matches with `sensor["type"] == "temperature_core"`, now it matches with `sensor["type"].startswith("temperature")` (With internal sensor type changing from `"temperature_core"` to `"temperature"`, `homeassistant/components/glances/const.py:39`. This type is just an internal representation, it does not affect the sensor type in homeassistant core or UI.)

Another example for `sensor_type` as `"memory_use"`, the previous matches `sensor["type"] == "memory_use"` and now it matches `sensor["type"].startswith("memory_use")` (Without any change to the sensor type).

Reply:
Oh I guess I see the problem, are you talking about an example like this:

For `memory_use` and `memory_use_percent`, this new `if sensor["type"].startswith(sensor_type):` will match them twice?

Reply:
I just revert the matching rule update and use an additional sensor type to discover HDD-related sensors.

Reply:
I updated this PR with fewer changes and this matching rule is not changed in the newer version.